### PR TITLE
RCIAM-472_Enable_send_Petition_Confirmation_Email_via_COmanage_REST

### PR DESCRIPTION
**Pull request by ioigoume**
## Enable `/co_petitions/resend` action via REST API
Request to resend the invitation for the CO Petition with ID `1234`

### Example Request
`curl -u "user:password" "https://example.com/registry/co_petitions/resend/1234.json" | python -m json.tool`
### Example Response
```bash
{
    "CoPetitions": {
        "recipient": "john.doe@example.com",
        "status": "Invitation sent to john.doe@example.com"
    },
    "ResponseType": "CoPetitions",
    "Version": "1.0"
}
```

### CODE REVIEW

File: app/Controller/CoPetitionsController.php [hunk 1]
```diff
@@ -2038,10 +2045,16 @@ function isAuthorized() {
       $args['contain'] = false;
       
       $pt = $this->CoPetition->find('first', $args);
-      
+
       if(!$pt) {
-        $this->Flash->set(_txt('er.notfound', array(_txt('ct.co_petitions.1', $petitionId))), array('key' => 'error'));
-        $this->redirect('/');
+        if($this->request->is('restful')) {
+          $this->Api->restResultHeader(404, "Not Found");
```
Comment (lionick) [inline app/Controller/CoPetitionsController.php]: may add something like in the line 2055 _txt('er.notfound', array(_txt('ct.co_petitions.1', $petitionId))) ?

File: app/View/CoPetitions/json/resend.ctp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * @license       Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ */
+
+// Get a pointer to our model
+$model = $this->name;                     // Unlike in the controllers, this is CamelPlural
+$req = Inflector::singularize($model);
+$modelpl = Inflector::tableize($req);
+$modelcc = $model;
+
+$response = array();
```
Comment (lionick) [inline app/View/CoPetitions/json/resend.ctp]: maybe this can be done
$response = (!empty($vv_response) ? $vv_response : array())
?

**Review by lionick — CHANGES_REQUESTED** [review]

File: app/Controller/CoPetitionsController.php (hunk 1 above)
Comment (ioigoume) [reply app/Controller/CoPetitionsController.php]: Done

File: app/View/CoPetitions/json/resend.ctp (hunk 2 above)
Comment (ioigoume) [reply app/View/CoPetitions/json/resend.ctp]: Done

**Comment by lionick** [comment]
LGTM

**Pull request merged**